tensorboard/webapp/webapp_data_source/tb_server_data_source_test.ts

```diff
@@ -68,6 +69,23 @@ describe('tb_server_data_source', () => {
           'data/plugins_listing?experimentalPlugin=foo&experimentalPlugin=bar'
         );
       });
+
+      it('handles "data/plugins_listing" failures', fakeAsync(() => {
+        const results = jasmine.createSpy();
+        const error = jasmine.createSpy();
+        dataSource.fetchPluginsListing([]).subscribe(results, error);
+
+        httpMock
+          .expectOne('data/plugins_listing')
+          .error(new ErrorEvent('FakeError'), {status: 501});
+        // Flush the promise in the microtask.
```

Comment:
Just wondering, do we make use of Promise in the data source?

Reply:
Not explicitly.